xla/service/cpu/onednn_ops_rewriter.cc

```diff
@@ -61,11 +63,35 @@ HloInstruction* FindLayerNormShift(HloInstruction* instr) {
   return shift;
 }
 
-std::optional<HloInstruction*> MatchSoftmax(HloInstruction* instr) {
+bool is_neg_inf_const_scalar(const HloInstruction* const_instr) {
+  if (!ShapeUtil::IsEffectiveScalar(const_instr->shape())) return false;
+  auto value = LiteralUtil::GetFirstScalarLiteral(const_instr->literal());
+  return literal_comparison::Equal(
+             value, LiteralUtil::MinValue(const_instr->shape().element_type()))
+      .ok();
+}
+
+auto MaxReduce(HloInstruction** instr) {
+  auto is_valid_reduce_max = [](const HloInstruction* reduce) {
+    HloComputation* reducer = reduce->to_apply();
+    return (reducer->root_instruction()->opcode() == HloOpcode::kMaximum) &&
```

Comment:
Could you elaborate more on this comment ? The is_valid_reduce_max looks precise enough ,as in, it checks for Maximum Computation and 2nd input as constant scalar and is negative infinity.

Reply:
My point is that you don't really check for a maximum computation here: you check that the root of the reducer computation is `maximum`, but it would match correctly a reduction computation such as e.g.

```
x = parameter(0)
y = parameter(1)
min_0 = minimum(x, y)
min_1 = minimum(y, x)
ROOT max = maximum(min_0, min_1)
```

which, would not be a `maximum` computation, would involve a `maximum` root, but still obey the monoid property (and hence is valid HLO to write).

Thus, I think your check for the reducer should be stricter, and ensure that the parameters to the `maximum` root are the parameters of the reducer computation---in any order.

Does this make sense?

Reply:
Yes you are right, thank you for catching it and explaining it. Agreed that it needs to have more precise check, I modified the code accordingly, hope thats fine